eodatasets3/scripts/tostac.py

```diff
@@ -2,75 +2,214 @@
 Convert a new-style ODC metadata doc to a Stac Item. (BETA/Incomplete)
 """
 import json
+import math
 from datetime import datetime
-from functools import partial
 from pathlib import Path
-from typing import Iterable
+from typing import Iterable, Dict
 from uuid import UUID
+from urllib.parse import urljoin
 
 import click
-import pyproj
-from click import echo
-from shapely.geometry.base import BaseGeometry
-from shapely.ops import transform
+import requests
+from jsonschema import validate
 
 from eodatasets3 import serialise
+from eodatasets3.model import DatasetDoc
 from eodatasets3.ui import PathPath
+from datacube.utils.geometry import Geometry, CRS
+
+
+# Mapping between EO3 field names and STAC properties object field names
+MAPPING_EO3_TO_STAC = {
+    "dtr:end_datetime": "end_datetime",
+    "dtr:start_datetime": "start_datetime",
+    "eo:gsd": "gsd",
+    "eo:instrument": "instruments",
+    "eo:platform": "platform",
+    "eo:constellation": "constellation",
+    "eo:off_nadir": "view:off_nadir",
+    "eo:azimuth": "view:azimuth",
+    "eo:sun_azimuth": "view:sun_azimuth",
+    "eo:sun_elevation": "view:sun_elevation",
+}
+
+
+def convert_value_to_stac_type(key: str, value):
+    """
+    Convert return type as per STAC specification
+    """
+    # In STAC spec, "instruments" have [String] type
+    if key == "eo:instrument":
+        return [value]
+    else:
+        return value
 
 
 @click.command(help=__doc__)
+@click.option(
+    "--stac-template",
+    "-t",
+    type=str,
```

Comment:
Use type `PathPath`, like the other file path options.

Reply:
Removed `--stac-template` options